`tests/kernel/device`: build fails for `hifive_unmatched`

**Describe the bug**
Trying to build the `test/kernel/device` test (any variant) for the `hifive_unmatched` target results in a build failure:

```
Cmake build failure
Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/tests/kernel/device
-- CMake version: 3.22.1
-- Found Python3: /usr/bin/python3 (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter 
-- Cache files will be written to: /root/.cache/zephyr
-- Zephyr version: 3.7.99 (/__w/zephyr/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: hifive_unmatched, qualifiers: fu740/s7
-- Found host-tools: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0)
-- Found toolchain: zephyr 0.17.0 (/opt/toolchains/zephyr-sdk-0.17.0)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.17.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/sifive/hifive_unmatched/hifive_unmatched_s7.dts
-- Found devicetree overlay: /__w/zephyr/zephyr/tests/kernel/device/boards/hifive_unmatched_fu740_s7.overlay
devicetree error: <Node /fakedomain_2 in '/__w/zephyr/zephyr/misc/empty_file.c'> lacks #power-domain-cells
CMake Error at /__w/zephyr/zephyr/cmake/modules/dts.cmake:301 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  /__w/zephyr/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  /__w/zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /__w/zephyr/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
```

First discovered in #77883.

**To Reproduce**
1. `west build -p -b hifive_unmatched tests/kernel/device`
2. see error

**Expected behavior**
Build ends successfully.

**Impact**
Showstopper.

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA or Version used: `9f93dede369c365c68b5618844e3d31e35327bc2`

--- a/tests/kernel/device/boards/hifive_unmatched.overlay
+++ b/tests/kernel/device/boards/hifive_unmatched.overlay
@@ -66,17 +66,20 @@
 	fakedomain_0: fakedomain_0 {
 		compatible = "fakedomain";
 		status = "okay";
+		#power-domain-cells = <0>;
 		power-domains = <&fakedomain_2>;
 	};
 
 	fakedomain_1: fakedomain_1 {
 		compatible = "fakedomain";
 		status = "okay";
+		#power-domain-cells = <0>;
 		power-domains = <&fakedomain_0>;
 	};
 
 	fakedomain_2: fakedomain_2 {
 		compatible = "fakedomain";
 		status = "okay";
+		#power-domain-cells = <0>;
 	};
 };

